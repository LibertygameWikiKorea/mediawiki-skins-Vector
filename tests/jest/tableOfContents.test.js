// @ts-nocheck
const tableOfContents = require( '../../resources/skins.vector.es6/tableOfContents.js' );

const template = `
    <ul>

        <li id="toc-foo" class="sidebar-toc-level-1">
            <a href="#foo">foo</a>
        </li>

        <li id="toc-bar" class="sidebar-toc-level-1">
            <a href="#bar">bar</a>
            <ul>
                <li id="toc-baz">
                    <a href="#baz">baz</a>
                </li>
            </ul>
        </li>

        <li id="toc-qux" class="sidebar-toc-level-1">
            <a href="#qux">qux</a>
        </li>

    </ul>
`;

let toc, fooSection, barSection, bazSection, quxSection;

beforeEach( () => {
	document.body.innerHTML = template;
	toc = tableOfContents( { container: document.body } );
	fooSection = document.getElementById( 'toc-foo' );
	barSection = document.getElementById( 'toc-bar' );
	bazSection = document.getElementById( 'toc-baz' );
	quxSection = document.getElementById( 'toc-qux' );
} );

test( 'Table of contents changes the active sections', () => {
	toc.changeActiveSection( 'toc-foo' );
	expect(
		fooSection.classList.contains( toc.ACTIVE_SECTION_CLASS ) &&
		!barSection.classList.contains( toc.ACTIVE_SECTION_CLASS ) &&
		!bazSection.classList.contains( toc.ACTIVE_SECTION_CLASS ) &&
		!quxSection.classList.contains( toc.ACTIVE_SECTION_CLASS )
	).toEqual( true );

	toc.changeActiveSection( 'toc-bar' );
	expect(
		!fooSection.classList.contains( toc.ACTIVE_SECTION_CLASS ) &&
		barSection.classList.contains( toc.ACTIVE_SECTION_CLASS ) &&
		!bazSection.classList.contains( toc.ACTIVE_SECTION_CLASS ) &&
		!quxSection.classList.contains( toc.ACTIVE_SECTION_CLASS )
	).toEqual( true );

	toc.changeActiveSection( 'toc-baz' );
	expect(
		!fooSection.classList.contains( toc.ACTIVE_SECTION_CLASS ) &&
		barSection.classList.contains( toc.ACTIVE_SECTION_CLASS ) &&
		bazSection.classList.contains( toc.ACTIVE_SECTION_CLASS ) &&
		!quxSection.classList.contains( toc.ACTIVE_SECTION_CLASS )
	).toEqual( true );

	toc.changeActiveSection( 'toc-qux' );
	expect(
		!fooSection.classList.contains( toc.ACTIVE_SECTION_CLASS ) &&
		!barSection.classList.contains( toc.ACTIVE_SECTION_CLASS ) &&
		!bazSection.classList.contains( toc.ACTIVE_SECTION_CLASS ) &&
		quxSection.classList.contains( toc.ACTIVE_SECTION_CLASS )
	).toEqual( true );
} );

test( 'Table of contents expands sections', () => {
	toc.expandSection( 'toc-foo' );
	expect(
		fooSection.classList.contains( toc.EXPANDED_SECTION_CLASS ) &&
		!barSection.classList.contains( toc.EXPANDED_SECTION_CLASS ) &&
		!bazSection.classList.contains( toc.EXPANDED_SECTION_CLASS ) &&
		!quxSection.classList.contains( toc.EXPANDED_SECTION_CLASS )
	).toEqual( true );

	toc.expandSection( 'toc-bar' );
	expect(
		fooSection.classList.contains( toc.EXPANDED_SECTION_CLASS ) &&
		barSection.classList.contains( toc.EXPANDED_SECTION_CLASS ) &&
		!bazSection.classList.contains( toc.EXPANDED_SECTION_CLASS ) &&
		!quxSection.classList.contains( toc.EXPANDED_SECTION_CLASS )
	).toEqual( true );
} );

test( 'Table of contents toggles expanded sections', () => {
	toc.toggleExpandSection( 'toc-foo' );
	expect(
		fooSection.classList.contains( toc.EXPANDED_SECTION_CLASS )
	).toEqual( true );

	toc.toggleExpandSection( 'toc-foo' );
	expect(
		fooSection.classList.contains( toc.EXPANDED_SECTION_CLASS )
	).toEqual( false );
} );
