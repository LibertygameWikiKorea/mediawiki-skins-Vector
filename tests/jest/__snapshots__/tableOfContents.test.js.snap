// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table of contents re-rendering re-renders toc when wikipage.tableOfContents hook is fired with empty sections 1`] = `
"<nav id=\\"mw-panel-toc\\" class=\\"sidebar-toc\\" role=\\"navigation\\" aria-labelledby=\\"sidebar-toc-label\\" data-event-name=\\"ui.sidebar-toc\\"></nav>
"
`;

exports[`Table of contents re-rendering re-renders toc when wikipage.tableOfContents hook is fired with sections 1`] = `
"<nav id=\\"mw-panel-toc\\" class=\\"sidebar-toc\\" role=\\"navigation\\" aria-labelledby=\\"sidebar-toc-label\\" data-event-name=\\"ui.sidebar-toc\\"><nav id=\\"mw-panel-toc\\" class=\\"sidebar-toc\\" role=\\"navigation\\" aria-labelledby=\\"sidebar-toc-label\\" data-event-name=\\"ui.sidebar-toc\\">
	<div id=\\"sidebar-toc-label\\" class=\\"sidebar-toc-header\\">
		<p class=\\"sidebar-toc-title\\">
		Contents
		<button class=\\"vector-toc-uncollapse-button\\">move to sidebar</button>
		<button class=\\"vector-toc-collapse-button\\">hide</button>
		</p>
	</div>
	<ul class=\\"sidebar-toc-contents\\" id=\\"mw-panel-toc-list\\">
		<li id=\\"toc-mw-content-text\\" class=\\"sidebar-toc-list-item sidebar-toc-level-1\\">
			<a href=\\"#\\" class=\\"sidebar-toc-link\\">
				<div class=\\"sidebar-toc-text\\">Beginning</div>
			</a>
		</li>
<li id=\\"toc-foo\\" class=\\"sidebar-toc-list-item sidebar-toc-level-1 sidebar-toc-list-item-expanded\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#foo\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">1</span>foo</div>
	</a>
	<ul id=\\"toc-foo-sublist\\" class=\\"sidebar-toc-list\\">
	</ul>
</li>
<li id=\\"toc-bar\\" class=\\"sidebar-toc-list-item sidebar-toc-level-1 sidebar-toc-list-item-expanded\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#bar\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">2</span>bar</div>
	</a>
		<button aria-controls=\\"toc-bar-sublist\\" class=\\"mw-ui-icon mw-ui-icon-wikimedia-expand mw-ui-icon-small sidebar-toc-toggle\\" aria-expanded=\\"true\\">
			Toggle bar subsection
		</button>
	<ul id=\\"toc-bar-sublist\\" class=\\"sidebar-toc-list\\">
<li id=\\"toc-baz\\" class=\\"sidebar-toc-list-item sidebar-toc-level-2\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#baz\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">2.1</span>baz</div>
	</a>
	<ul id=\\"toc-baz-sublist\\" class=\\"sidebar-toc-list\\">
<li id=\\"toc-qux\\" class=\\"sidebar-toc-list-item sidebar-toc-level-3\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#qux\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">2.1.1</span>qux</div>
	</a>
	<ul id=\\"toc-qux-sublist\\" class=\\"sidebar-toc-list\\">
	</ul>
</li>
	</ul>
</li>
	</ul>
</li>
<li id=\\"toc-quux\\" class=\\"sidebar-toc-list-item sidebar-toc-level-1 sidebar-toc-list-item-expanded\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#quux\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">3</span>quux</div>
	</a>
	<ul id=\\"toc-quux-sublist\\" class=\\"sidebar-toc-list\\">
	</ul>
</li>
<li id=\\"toc-bat\\" class=\\"sidebar-toc-list-item sidebar-toc-level-1 sidebar-toc-list-item-expanded\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#bat\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">4</span>bat</div>
	</a>
	<ul id=\\"toc-bat-sublist\\" class=\\"sidebar-toc-list\\">
	</ul>
</li>
	</ul>
</nav>
</nav>
"
`;

exports[`Table of contents renders when \`vector-is-collapse-sections-enabled\` is false 1`] = `
"<nav id=\\"mw-panel-toc\\" class=\\"sidebar-toc\\" role=\\"navigation\\" aria-labelledby=\\"sidebar-toc-label\\" data-event-name=\\"ui.sidebar-toc\\">
	<div id=\\"sidebar-toc-label\\" class=\\"sidebar-toc-header\\">
		<p class=\\"sidebar-toc-title\\">
		Contents
		<button class=\\"vector-toc-uncollapse-button\\"></button>
		<button class=\\"vector-toc-collapse-button\\"></button>
		</p>
	</div>
	<ul class=\\"sidebar-toc-contents\\" id=\\"mw-panel-toc-list\\">
		<li id=\\"toc-mw-content-text\\" class=\\"sidebar-toc-list-item sidebar-toc-level-1\\">
			<a href=\\"#\\" class=\\"sidebar-toc-link\\">
				<div class=\\"sidebar-toc-text\\">Beginning</div>
			</a>
		</li>
<li id=\\"toc-foo\\" class=\\"sidebar-toc-list-item sidebar-toc-level-1 sidebar-toc-list-item-expanded\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#foo\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">1</span>foo</div>
	</a>
	<ul id=\\"toc-foo-sublist\\" class=\\"sidebar-toc-list\\">
	</ul>
</li>
<li id=\\"toc-bar\\" class=\\"sidebar-toc-list-item sidebar-toc-level-1 sidebar-toc-list-item-expanded\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#bar\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">2</span>bar</div>
	</a>
		<button aria-controls=\\"toc-bar-sublist\\" class=\\"mw-ui-icon mw-ui-icon-wikimedia-expand mw-ui-icon-small sidebar-toc-toggle\\" aria-expanded=\\"true\\">
			Toggle bar subsection
		</button>
	<ul id=\\"toc-bar-sublist\\" class=\\"sidebar-toc-list\\">
<li id=\\"toc-baz\\" class=\\"sidebar-toc-list-item sidebar-toc-level-2\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#baz\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">2.1</span>baz</div>
	</a>
	<ul id=\\"toc-baz-sublist\\" class=\\"sidebar-toc-list\\">
<li id=\\"toc-qux\\" class=\\"sidebar-toc-list-item sidebar-toc-level-3\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#qux\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">2.1.1</span>qux</div>
	</a>
	<ul id=\\"toc-qux-sublist\\" class=\\"sidebar-toc-list\\">
	</ul>
</li>
	</ul>
</li>
	</ul>
</li>
<li id=\\"toc-quux\\" class=\\"sidebar-toc-list-item sidebar-toc-level-1 sidebar-toc-list-item-expanded\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#quux\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">3</span>quux</div>
	</a>
	<ul id=\\"toc-quux-sublist\\" class=\\"sidebar-toc-list\\">
	</ul>
</li>
	</ul>
</nav>
"
`;

exports[`Table of contents renders when \`vector-is-collapse-sections-enabled\` is true 1`] = `
"<nav id=\\"mw-panel-toc\\" class=\\"sidebar-toc\\" role=\\"navigation\\" aria-labelledby=\\"sidebar-toc-label\\" data-event-name=\\"ui.sidebar-toc\\">
	<div id=\\"sidebar-toc-label\\" class=\\"sidebar-toc-header\\">
		<p class=\\"sidebar-toc-title\\">
		Contents
		<button class=\\"vector-toc-uncollapse-button\\"></button>
		<button class=\\"vector-toc-collapse-button\\"></button>
		</p>
	</div>
	<ul class=\\"sidebar-toc-contents\\" id=\\"mw-panel-toc-list\\">
		<li id=\\"toc-mw-content-text\\" class=\\"sidebar-toc-list-item sidebar-toc-level-1\\">
			<a href=\\"#\\" class=\\"sidebar-toc-link\\">
				<div class=\\"sidebar-toc-text\\">Beginning</div>
			</a>
		</li>
<li id=\\"toc-foo\\" class=\\"sidebar-toc-list-item sidebar-toc-level-1\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#foo\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">1</span>foo</div>
	</a>
	<ul id=\\"toc-foo-sublist\\" class=\\"sidebar-toc-list\\">
	</ul>
</li>
<li id=\\"toc-bar\\" class=\\"sidebar-toc-list-item sidebar-toc-level-1\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#bar\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">2</span>bar</div>
	</a>
		<button aria-controls=\\"toc-bar-sublist\\" class=\\"mw-ui-icon mw-ui-icon-wikimedia-expand mw-ui-icon-small sidebar-toc-toggle\\" aria-expanded=\\"false\\">
			Toggle bar subsection
		</button>
	<ul id=\\"toc-bar-sublist\\" class=\\"sidebar-toc-list\\">
<li id=\\"toc-baz\\" class=\\"sidebar-toc-list-item sidebar-toc-level-2\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#baz\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">2.1</span>baz</div>
	</a>
	<ul id=\\"toc-baz-sublist\\" class=\\"sidebar-toc-list\\">
<li id=\\"toc-qux\\" class=\\"sidebar-toc-list-item sidebar-toc-level-3\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#qux\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">2.1.1</span>qux</div>
	</a>
	<ul id=\\"toc-qux-sublist\\" class=\\"sidebar-toc-list\\">
	</ul>
</li>
	</ul>
</li>
	</ul>
</li>
<li id=\\"toc-quux\\" class=\\"sidebar-toc-list-item sidebar-toc-level-1\\">
	<a class=\\"sidebar-toc-link\\" href=\\"#quux\\">
		<div class=\\"sidebar-toc-text\\">
		<span class=\\"sidebar-toc-numb\\">3</span>quux</div>
	</a>
	<ul id=\\"toc-quux-sublist\\" class=\\"sidebar-toc-list\\">
	</ul>
</li>
	</ul>
</nav>
"
`;
